etna/search/forms.py

```diff
@@ -1,33 +1,46 @@
+import re
+
+from typing import Dict, List, Tuple, Union
+
 from django import forms
 from django.core.validators import MinLengthValidator
+from django.utils.functional import cached_property
 
 from ..ciim.client import SortBy, SortOrder
+from ..ciim.constants import COLLECTION_CHOICES, LEVEL_CHOICES
 
 
 class DynamicMultipleChoiceField(forms.MultipleChoiceField):
-    """MultipleChoiceField whose choices are populated by API response data.
-
-    Valid filter options are returned by the API as aggregation data belong to
-    a result set.
+    """MultipleChoiceField whose choices can be updated to reflect API response data."""
 
-    This field populates its choice list from aggregation data. This means that
-    at the point of form validation, this field's choices are empty.
-    """
-
-    def __init__(self, filter_key, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def valid_value(self, value):
+        """Disable validation if the field choices are not yet set."""
+        if not self.choices:
+            return True
+        return super().valid_value(value)
 
-        # prefix when passing to filter_aggregations. Format <prefix>:<key>
-        self.filter_key = filter_key
+    @cached_property
+    def original_choice_labels(self):
+        return {value: label for value, label in self.choices}
 
-    def valid_value(self, value):
-        """Disable validation, field doesn't have choices until the API is called."""
-        return True
+    def choice_label_from_api_data(self, data: Dict[str, Union[str, int]]) -> str:
+        count = f"{data['doc_count']:,}"
+        try:
+            return f"{data['label']} ({count})"
+        except KeyError:
+            pass
```

Comment:
Although it does not currently, it would make sense for API to provide separate `key` and `label` fields in the response - in which case, we should respect that label. For now, this will always fail and the label will be derived from existing choice labels or the `key` value